kernel/src/net/socket/netlink/route/bound.rs

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::ops::Sub;
+
+use super::message::RtnlMessage;
+use crate::{
+    events::IoEvents,
+    net::socket::{
+        netlink::{
+            message::ProtocolSegment, route::kernel::get_netlink_route_kernel, table::BoundHandle,
+            NetlinkSocketAddr,
+        },
+        SendRecvFlags,
+    },
+    prelude::*,
+    util::{MultiRead, MultiWrite},
+};
+
+pub(super) struct BoundNetlinkRoute {
+    handle: BoundHandle,
+    receive_queue: Mutex<VecDeque<RtnlMessage>>,
+}
+
+impl BoundNetlinkRoute {
+    pub(super) const fn new(handle: BoundHandle) -> Self {
+        Self {
+            handle,
+            receive_queue: Mutex::new(VecDeque::new()),
+        }
+    }
+
+    pub(super) const fn addr(&self) -> NetlinkSocketAddr {
+        self.handle.addr()
+    }
+
+    pub(super) fn try_send(
+        &self,
+        reader: &mut dyn MultiRead,
+        remote: Option<&NetlinkSocketAddr>,
+        flags: SendRecvFlags,
+    ) -> Result<usize> {
+        // TODO: Deal with flags
+        if !flags.is_all_supported() {
+            warn!("unsupported flags: {:?}", flags);
+        }
+
+        if let Some(remote) = remote {
+            // TODO: Further check whether other socket address can be supported.
+            if *remote != NetlinkSocketAddr::new_unspecified() {
+                return_errno_with_message!(
+                    Errno::ECONNREFUSED,
+                    "sending netlink route messages to user space is not supported"
+                );
+            }
+        } else {
+            // TODO: We should use the connected remote address, if any.
+        }
+
+        let mut nlmsg = {
+            let sum_lens = reader.sum_lens();
+
+            match RtnlMessage::read_from(reader) {
+                Ok(nlmsg) => nlmsg,
+                Err(e) if e.error() == Errno::EFAULT => {
+                    // EFAULT indicates an error occurred while copying data from user space,
+                    // and this error should be returned back to user space.
+                    return Err(e);
+                }
+                Err(e) => {
+                    // Errors other than EFAULT indicate a failure in parsing the netlink message.
+                    // These errors should be silently ignored.
+                    warn!("failed to send netlink message: {:?}", e);
+                    return Ok(sum_lens);
+                }
+            }
+        };
+
+        let local_port = self.addr().port();
+        for segment in nlmsg.segments_mut() {
+            let header = segment.header_mut();
+            if header.pid == 0 {
+                header.pid = local_port;
```

Comment:
Could you add some comments to describe this behavior? I'm not sure if we should change the input segments or pass an extra argument to `get_netlink_route_kernel().request`.